src/inekf_odometry.cpp

```diff
@@ -0,0 +1,231 @@
+
+class InEKFNode : public rclcpp::Node
+{
+public:
+    explicit InEKFNode()
+    : Node("inekf")
+    {
+        this->declare_parameter("base_frame", rclcpp::PARAMETER_STRING);
+        this->declare_parameter("odom_frame", rclcpp::PARAMETER_STRING);
+
+        rmw_qos_profile_t qos = rmw_qos_profile_sensor_data;
+        qos.reliability = RMW_QOS_POLICY_RELIABILITY_RELIABLE;
+        qos.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+        
+        // Write subscribers to feet odometry and IMU
+        pos_feet_subscriber_ = this->create_subscription<go2_odometry::msg::OdometryVector>(
+            "odometry/feet_pos", 
+            rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(qos), qos), 
+            std::bind(&InEKFNode::listener_feet_callback, this, std::placeholders::_1)
+        );
+        
+        imu_subscriber_ = this->create_subscription<sensor_msgs::msg::Imu>(
+            "imu", 
+            rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(qos), qos), 
+            std::bind(&InEKFNode::listener_callback, this, std::placeholders::_1)
+        );
+        
+        // Write publishers and TF
+        odometry_publisher_ = this->create_publisher<nav_msgs::msg::Odometry>(
+            "odometry/filtered", 
+            rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(qos), qos)
+        ); 
+
+        
+        tf_broadcaster_ =
+            std::make_unique<tf2_ros::TransformBroadcaster>(*this);
+        geometry_msgs::msg::TransformStamped t;
+
+        // Robot-related quantities
+        foot_frame_names_ = {"FL_foot", "FR_foot", "RL_foot", "RR_foot"};
+        foot_frame_ids_ = {0, 1, 2, 3};
+
+        // Time variables 
+        DT_MIN_ = 1e-6;
+        DT_MAX_ = 1;
+        dt_ = 0;
+        t_ = 0;
+        t_prev_ = 0;
+        imu_measurement_.resize(6);
+        imu_measurement_prev_.resize(6);
+
+        // Setup filter
+        inekf::RobotState initial_state;
+        // Initialize state mean
+        Eigen::Matrix3d R0;
+        Eigen::Vector3d v0, p0, bg0, ba0, gravity;
+        R0 << 1, 0, 0, // initial orientation
+            0, -1, 0,  // IMU frame is rotated 90deg about the x-axis
+            0, 0, -1;
+        v0 << 0, 0, 0;  // initial velocity
+        p0 << 0, 0, 0.07;  // initial position
+        bg0 << 0, 0, 0; // initial gyroscope bias
+        ba0 << 0, 0, 0; // initial accelerometer bias
+        gravity << 0, 0, -9.81; 
+
+        initial_state.setRotation(R0);
+        initial_state.setVelocity(v0);
+        initial_state.setPosition(p0);
+        initial_state.setGyroscopeBias(bg0);
+        initial_state.setAccelerometerBias(ba0);
+
+        // Initialize state covariance
+        inekf::NoiseParams noise_params;
+        noise_params.setGyroscopeNoise(0.01);
+        noise_params.setAccelerometerNoise(0.1);
+        noise_params.setGyroscopeBiasNoise(0.00001);
+        noise_params.setAccelerometerBiasNoise(0.0001);
+        noise_params.setContactNoise(0.01);
```

Comment:
It would be nice to document those values unit and put it in a config file or a ros parameters